--- a/tools/Makefile	2018-10-17 17:54:24.784387325 +0500
+++ b/tools/Makefile	2018-10-17 17:54:30.484387187 +0500
@@ -122,7 +122,7 @@
 tools/_libfdt.so: $(patsubst %.o,%.c,$(LIBFDT_OBJS)) tools/libfdt_wrap.c
 	LDFLAGS="$(HOSTLDFLAGS)" python $(srctree)/lib/libfdt/setup.py \
 		"$(_hostc_flags)" $^
-	mv _libfdt.so $@
+	mv _libfdt*.so $@
 
 tools/libfdt_wrap.c: $(srctree)/lib/libfdt/libfdt.swig
 	swig -python -o $@ $<
