--- a/arch/arm/boot/dts/imx6ull.dtsi	2017-09-07 14:46:46.000000000 +0500
+++ b/arch/arm/boot/dts/imx6ull.dtsi	2017-10-27 15:44:27.198578759 +0500
@@ -1,6 +1,5 @@
 /*
  * Copyright 2015-2016 Freescale Semiconductor, Inc.
- * Copyright 2017 NXP.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
@@ -58,16 +57,12 @@
 			clock-latency = <61036>; /* two CLK32 periods */
 			operating-points = <
 				/* kHz	uV */
-				996000	1275000
-				792000	1225000
 				528000	1175000
 				396000	1025000
 				198000	950000
 			>;
 			fsl,soc-operating-points = <
 				/* KHz	uV */
-				996000	1175000
-				792000	1175000
 				528000	1175000
 				396000	1175000
 				198000	1175000
@@ -86,8 +81,6 @@
 				 <&clks IMX6UL_CLK_OSC>;
 			clock-names = "arm", "pll2_bus",  "pll2_pfd2_396m", "secondary_sel", "step",
 				      "pll1_sw", "pll1_sys", "pll1_bypass", "pll1", "pll1_bypass_src", "osc";
-			arm-supply = <&reg_arm>;
-			soc-supply = <&reg_soc>;
 		};
 	};
 
@@ -249,7 +242,7 @@
 						      "rxtx1", "rxtx2",
 						      "rxtx3", "rxtx4",
 						      "rxtx5", "rxtx6",
-						      "rxtx7", "spba";
+						      "rxtx7", "dma";
 					status = "disabled";
 				};
 
@@ -343,7 +336,7 @@
 						 <&clks IMX6UL_CLK_ESAI_IPG>,
 						 <&clks IMX6UL_CLK_SPBA>;
 					clock-names = "core", "mem", "extal",
-						      "fsys", "spba";
+						      "fsys", "dma";
 					dmas = <&sdma 0 21 0>, <&sdma 47 21 0>;
 					dma-names = "rx", "tx";
 					dma-source = <&gpr 0 14 0 15>;
@@ -410,7 +403,7 @@
 						"asrck_1", "asrck_2", "asrck_3", "asrck_4",
 						"asrck_5", "asrck_6", "asrck_7", "asrck_8",
 						"asrck_9", "asrck_a", "asrck_b", "asrck_c",
-						"asrck_d", "asrck_e", "asrck_f", "spba";
+						"asrck_d", "asrck_e", "asrck_f", "dma";
 					dmas = <&sdma 17 23 1>, <&sdma 18 23 1>, <&sdma 19 23 1>,
 						<&sdma 20 23 1>, <&sdma 21 23 1>, <&sdma 22 23 1>;
 					dma-names = "rxa", "rxb", "rxc",
@@ -499,8 +492,8 @@
 				reg = <0x02098000 0x4000>;
 				interrupts = <GIC_SPI 55 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&clks IMX6UL_CLK_GPT1_BUS>,
-					 <&clks IMX6UL_CLK_GPT_3M>;
-				clock-names = "ipg", "osc_per";
+					 <&clks IMX6UL_CLK_GPT1_SERIAL>;
+				clock-names = "ipg", "per";
 			};
 
 			gpio1: gpio@0209c000 {
@@ -723,7 +716,7 @@
 					regmap = <&snvs>;
 					interrupts = <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
 					linux,keycode = <KEY_POWER>;
-					wakeup-source;
+					wakeup;
 				};
 			};
 
